packages/thirdweb/src/react/web/ui/prebuilt/NFTProvider/useNFTMedia.ts

```diff
@@ -0,0 +1,100 @@
+import { useQuery } from "@tanstack/react-query";
+import type { Chain } from "../../../../../chains/types.js";
+import type { ThirdwebClient } from "../../../../../client/client.js";
+import { getContract } from "../../../../../contract/contract.js";
+import { punkImageSvg } from "../../../../../extensions/cryptopunks/__generated__/CryptoPunks/read/punkImageSvg.js";
+import { tokenURI } from "../../../../../extensions/erc721/__generated__/IERC721A/read/tokenURI.js";
+import { uri } from "../../../../../extensions/erc1155/__generated__/IERC1155/read/uri.js";
+import type { BaseTransactionOptions } from "../../../../../transaction/types.js";
+
+export function useNFTMedia(props: {
+  contractAddress?: string;
```

Comment:
Can use ThirdwebContract instance

Reply:
has to ;)

Reply:
fwiw I don't think we need this hook at all

Reply:
but we _could_ have it if we want